diff --git a/templates/page.tpl.php b/templates/page.tpl.php
index 0765a9c..2a504d5 100644
--- a/templates/page.tpl.php
+++ b/templates/page.tpl.php
@@ -26,7 +26,7 @@
   <?php endif; ?>
 
   <?php if ($secondary_local_tasks): ?>
-    <div class="tabs-secondary clearfix"><ul class="tabs secondary"><?php print render($secondary_local_tasks); ?></ul></div>
+    <div class="tabs-secondary clearfix"><?php print render($secondary_local_tasks); ?></div>
   <?php endif; ?>
 
 </div>
