diff --git a/.htaccess b/.htaccess
index 37fce10d32..bbd65099d8 100644
--- a/.htaccess
+++ b/.htaccess
@@ -44,8 +44,8 @@ DirectoryIndex index.php index.html index.htm
   # Enable expirations.
   ExpiresActive On
 
-  # Cache all files for 2 weeks after access (A).
-  ExpiresDefault A1209600
+  # Cache all files for 52 weeks after access (A).
+  ExpiresDefault A31449600
 
   <FilesMatch \.php$>
     # Do not allow PHP scripts to be cached unless they explicitly send cache
