diff --git a/includes/features.field.inc b/includes/features.field.inc
index 849081d..3421d0a 100644
--- a/includes/features.field.inc
+++ b/includes/features.field.inc
@@ -164,6 +164,11 @@ function field_base_features_export_render($module, $data, $export = NULL) {
         unset($field['storage']['details']);
       }
 
+      // DOMINO patch: We hard code field_permissions to 0 in the export,
+      // so that we can change them on a per-brand basis
+      // using HOOK_field_default_field_bases_alter().
+      $field['field_permissions']['type'] = 0;
+
       _field_instance_features_export_sort($field);
       $field_export = features_var_export($field, '  ');
       $field_prefix = '  // Exported field_base: ';
@@ -175,6 +180,7 @@ function field_base_features_export_render($module, $data, $export = NULL) {
       else {
         $code[] = $field_prefix . $field_identifier . '.';
       }
+      $code[] = "  // Field permissions hardcoded to 0. See features.field.inc.";
       $code[] = "  \$field_bases[{$field_identifier}] = {$field_export};";
       $code[] = "";
     }
