diff --git a/includes/admin.inc b/includes/admin.inc
index bf1ac46..d9fc08a 100644
--- a/includes/admin.inc
+++ b/includes/admin.inc
@@ -236,11 +236,13 @@ function _google_tag_snippets_save($realm_name = '', $realm_key = '') {
  * Stores data layer classes based on current settings.
  */
 function _google_tag_classes_save($realm_name = '', $realm_key = '') {
+  $data_layer = variable_get('google_tag_data_layer', 'dataLayer');
   if (module_exists('datalayer') && $data_layer == 'dataLayer') {
     // Save classes for data layer.
     // @todo How to or can we set the name of the data layer?
     // This is an open issue on datalayer project; after it is implemented in
     // datalayer then remove second condition on if block.
+    $classes = variable_get('google_tag_data_layer_classes', array());
     _google_tag_data_layer_snippet($classes);
     if ($realm_name) {
       variable_realm_set($realm_name, $realm_key, 'google_tag_data_layer_classes', $classes);
