diff --git a/disqus.module b/disqus.module
index 91d56b13..80915646 100644
--- a/disqus.module
+++ b/disqus.module
@@ -818,7 +818,7 @@ function disqus_theme() {
 function theme_disqus_noscript($variables = array()) {
   $disqus = $variables['disqus'];
   // Return the comment markup.
-  return '<noscript><p>' . l(t('View the discussion thread.'), 'http://' . $disqus['domain'] . '.disqus.com/?url=' . urlencode($disqus['url'])) . '</p></noscript>';
+  return '<noscript><p>' . l(t('View the discussion thread.'), 'https://' . $disqus['domain'] . '.disqus.com/?url=' . urlencode($disqus['url'])) . '</p></noscript>';
 }
