diff --git a/i18n_string/i18n_string.module b/i18n_string/i18n_string.module
index d184d7d..5f73c57 100644
--- a/i18n_string/i18n_string.module
+++ b/i18n_string/i18n_string.module
@@ -903,7 +903,9 @@ function i18n_string_object_translate($type, $object, $options = array()) {
  *   Object or array
  */
 function i18n_string_object_remove($type, $object, $options = array()) {
-  return i18n_object($type, $object)->strings_remove($options);
+  if(is_object($object) && method_exists($object, 'strings_remove')) {
+    return i18n_object($type, $object)->strings_remove($options);
+  }
 }
 
 /**
@@ -915,7 +917,9 @@ function i18n_string_object_remove($type, $object, $options = array()) {
  *   Object or array
  */
 function i18n_string_object_update($type, $object, $options = array()) {
-  return i18n_object($type, $object)->strings_update($options);
+  if(is_object($object) && method_exists($object, 'strings_update')) {
+    return i18n_object($type, $object)->strings_update($options);
+  }
 }
 
 /**
