diff --git a/leaflet_widget.module b/leaflet_widget.module
index b81a7d7..8544652 100644
--- a/leaflet_widget.module
+++ b/leaflet_widget.module
@@ -103,6 +103,9 @@ function leaflet_widget_field_widget_form(&$form, &$form_state, $field, $instanc
       $id = 'leaflet-widget_' . str_replace('_', '-', $instance['field_name']);
       $class = 'leaflet-widget';
       $style = 'height: 300px;';
+      
+      // Add counter and append to id.
+      $id .=  '-count' . ($element['#field_parents'][2] + 1);
 
       $settings['map']['widget'] = array();
       $settings['map']['widget']['attach'] = "$id-input";
