diff --git a/manualcrop.helpers.inc b/manualcrop.helpers.inc
index dcd38f6..0f3760e 100644
--- a/manualcrop.helpers.inc
+++ b/manualcrop.helpers.inc
@@ -804,12 +804,14 @@ function _manualcrop_image_style_label($style) {
  *   TRUE if this is a Manual Crop (cropping/reuse) effect, FALSE otherwise.
  */
 function _manualcrop_is_own_effect($effect, $crop_effect = NULL) {
-  if ($effect['module'] == 'manualcrop') {
-    if (is_null($crop_effect)) {
-      return TRUE;
-    };
+  if (isset($effect['module'])) {
+    if ($effect['module'] == 'manualcrop') {
+      if (is_null($crop_effect)) {
+        return TRUE;
+      };
 
-    return $crop_effect ^ in_array($effect['name'], array('manualcrop_reuse', 'manualcrop_auto_reuse'));
+      return $crop_effect ^ in_array($effect['name'], array('manualcrop_reuse', 'manualcrop_auto_reuse'));
+    }
   }
   return FALSE;
 }
