diff --git a/metatag_import_export_csv_download_admin.inc b/metatag_import_export_csv_download_admin.inc
index eef6115..505397f 100644
--- a/metatag_import_export_csv_download_admin.inc
+++ b/metatag_import_export_csv_download_admin.inc
@@ -103,6 +103,7 @@ function metatag_import_export_csv_process_csv($nid_list, $type) {
     'Meta Description',
     'Meta Abstract',
     ' Meta Keywords',
+    'Language',
   );
   fputcsv($fp, $header);
 
@@ -122,6 +123,7 @@ function metatag_import_export_csv_process_csv($nid_list, $type) {
         $meta_description,
         $meta_abstract,
         $meta_keywords,
+        $node->language,
       );
       file_put_contents($file_url, $row, FILE_APPEND);
       fputcsv($fp, $row);
