diff --git a/panopoly_wysiwyg.make b/panopoly_wysiwyg.make
index 50f5b69..4eb2891 100644
--- a/panopoly_wysiwyg.make
+++ b/panopoly_wysiwyg.make
@@ -36,11 +36,6 @@ projects[caption_filter][patch][2455253] = https://www.drupal.org/files/issues/c
 
 ; Include our Editors
 
-libraries[tinymce][download][type] = get
-libraries[tinymce][download][url] = http://download.moxiecode.com/tinymce/tinymce_3.5.11.zip
-libraries[tinymce][patch][1561882] = http://drupal.org/files/1561882-cirkuit-theme-tinymce-3.5.8.patch
-libraries[tinymce][patch][2876031] = https://www.drupal.org/files/issues/tinymce-chrome58-fix-2876031-5.patch
-
 libraries[markitup][download][type] = git
 libraries[markitup][download][url] = https://github.com/markitup/1.x.git
 libraries[markitup][download][revision] = 2c88c42
