diff --git token.pages.inc token.pages.inc
index 4fe3968..209aad7 100644
--- token.pages.inc
+++ token.pages.inc
@@ -132,6 +132,7 @@ function theme_token_tree($variables) {
     }
 
     if ($multiple_token_types) {
+      $type_info = !is_array($type_info) ? array() : $type_info;
       $row = _token_token_tree_format_row($type, $type_info, TRUE);
       unset($row['data']['value']);
       $rows[] = $row;
